src/main/java/net/ccbluex/liquidbounce/utils/render/RenderUtils.kt

```diff
@@ -358,34 +358,34 @@ object RenderUtils : MinecraftInstance() {
         glEnable(GL_LINE_SMOOTH)
         glLineWidth(width)
 
-        val tessellator = Tessellator.getInstance()
-        val worldrenderer = tessellator.worldRenderer
-        worldrenderer.begin(GL_LINE_LOOP, DefaultVertexFormats.POSITION_COLOR)
+        glColor4f(red, green, blue, alpha)
+        glBegin(GL_LINE_LOOP)
 
-        val degreeIncrement = PI / 180
-        val radiusF = radius.toDouble()
+        val radiusD = radius.toDouble()
 
         val corners = listOf(
-            Triple(newX2 - radiusF, newY2 - radiusF, 0),
-            Triple(newX2 - radiusF, newY1 + radiusF, 90),
-            Triple(newX1 + radiusF, newY1 + radiusF, 180),
-            Triple(newX1 + radiusF, newY2 - radiusF, 270)
+            Triple(newX2 - radiusD, newY2 - radiusD, 0.0),
+            Triple(newX2 - radiusD, newY1 + radiusD, 90.0),
+            Triple(newX1 + radiusD, newY1 + radiusD, 180.0),
+            Triple(newX1 + radiusD, newY2 - radiusD, 270.0)
         )
 
         for ((cx, cy, startAngle) in corners) {
-            for (i in 0..90) {
-                val angle = (startAngle + i) * degreeIncrement
-                val x = cx + radiusF * sin(angle)
-                val y = cy + radiusF * cos(angle)
-                worldrenderer.pos(x, y, 0.0).color(red, green, blue, alpha).endVertex()
+            for (i in 0..90 step 10) {
+                val angle = Math.toRadians(startAngle + i)
+                val x = cx + radiusD * sin(angle)
+                val y = cy + radiusD * cos(angle)
+                glVertex2d(x, y)
             }
         }
```

Comment:
Why is this an performance improvement?

Reply:
Not sure but using tesselator in my testing get less fps, well I'm assuming tesselator is better to be used in 3d then 2d rendering